### PR TITLE
feat(integrations): add support for langchain and langsmith

### PR DESCRIPTION
**Pull request by jjmachan**

### CODE REVIEW

**Comment by hinthornw** [comment]
@jjmachan happy to help if there's anything blocking this release! Would love to land so we can share more broadly (https://github.com/langchain-ai/langsmith-cookbook/blob/main/testing-examples/ragas/ragas.ipynb)

File: src/ragas/metrics/base.py [hunk 1]
```diff
@@ -25,6 +25,27 @@
...
+        keys = ["question", "contexts"]
+    elif eval_mod == EvaluationMode.gc:
+        keys = ["contexts", "ground_truth"]
+    elif eval_mod == EvaluationMode.ga:
+        keys = ["answer", "ground_truth"]
+    elif eval_mod == EvaluationMode.qga:
+        keys = ["question", "contexts", "answer", "ground_truth"]
+    elif eval_mod == EvaluationMode.qcg:
+        keys = ["question", "contexts", "ground_truth"]
+
```
Comment (mspronesti) [inline src/ragas/metrics/base.py:45]:
```suggestion
ignore_columns = ignore_columns or []
```

**Review by mspronesti — CHANGES_REQUESTED** [review]

File: src/ragas/metrics/base.py [hunk 2]
```diff
@@ -25,6 +25,27 @@
 EvaluationMode = Enum("EvaluationMode", "qac qa qc gc ga qga qcg")
 
 
+def get_required_columns(
+    eval_mod: EvaluationMode, ignore_columns: t.List[str] = []
```
Comment (mspronesti) [inline src/ragas/metrics/base.py:29]:
```suggestion
    eval_mod: EvaluationMode, ignore_columns: t.Oprional[t.List[str]] = None
```

**Comment by jjmachan** [comment]
thanks for the catch @mspronesti 🙌🏽 

ps: will add docs soon but feel free to look at https://notes.explodinggradients.com/en/latest/RAG-Experiments/basecamp-handbook/baseline-langchain.html for how to use this in action, I'll docs in the mean time

**Comment by mspronesti** [comment]
@jjmachan I accidentally misspelled `Optional` with an "r" 😅

**Comment by jjmachan** [comment]
no worries - I missed it but at least our linters caught it 😂

**Pull request merged**